From 492baf40be26c1ef321c7f00c22aa61825888b06 Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lb@semihalf.com>
Date: Wed, 17 Oct 2018 14:54:06 +0300
Subject: [PATCH 0305/1051] octeontx: change count threshold for vhaura

When FPA_VHAURA_CNT is equal or crosses FPA_VHAURA_CNT_THRESHOLD then
A_THRESH interrupt is generated. Previously FPA_VHAURA_CNT_THRESHOLD
was set to half of the numbers of buffers in a pool. This patch
changes FPA_VHAURA_CNT_THRESHOLD to a number of buffers in a pool
minus one.

Signed-off-by: Lukasz Bartosik <lb@semihalf.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
index 7bd3e72a8874..0335283ee2b8 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
@@ -228,7 +228,7 @@ static int fpa_vf_setup(struct fpavf *fpa, u64 num_buffers, u32 buf_len,
 		fpa_vf_do_test(fpa, num_buffers);
 
 	/*Setup THRESHOLD*/
-	fpavf_reg_write(fpa, FPA_VF_VHAURA_CNT_THRESHOLD(0), num_buffers / 2);
+	fpavf_reg_write(fpa, FPA_VF_VHAURA_CNT_THRESHOLD(0), num_buffers - 1);
 
 	return 0;
 }
-- 
2.17.1

